yarn-project/ethereum/src/utils.ts

```diff
@@ -64,3 +66,110 @@ function tryExtractEvent<
     }
   }
 }
+
+export function prettyLogViemErrorMsg(err: any) {
+  if (err instanceof BaseError) {
+    const revertError = err.walk(err => err instanceof ContractFunctionRevertedError);
+    if (revertError instanceof ContractFunctionRevertedError) {
+      const errorName = revertError.data?.errorName ?? '';
+      const args =
+        revertError.metaMessages && revertError.metaMessages?.length > 1 ? revertError.metaMessages[1].trimStart() : '';
+      return `${errorName}${args}`;
+    }
+  }
+  return err?.message ?? err;
+}
+
+export function formatViemError(error: any): string {
+  const truncateHex = (hex: string, length = 100) => {
+    if (!hex || typeof hex !== 'string') {
+      return hex;
+    }
+    if (!hex.startsWith('0x')) {
+      return hex;
+    }
+    if (hex.length <= length * 2) {
+      return hex;
+    }
+    return `${hex.slice(0, length)}...${hex.slice(-length)}`;
+  };
+
+  const formatRequestBody = (body: string) => {
+    try {
+      const parsed = JSON.parse(body);
+
+      // Recursively process all parameters that might contain hex strings
+      const processParams = (obj: any): any => {
+        if (Array.isArray(obj)) {
+          return obj.map(item => processParams(item));
+        }
+        if (typeof obj === 'object' && obj !== null) {
+          const result: any = {};
+          for (const [key, value] of Object.entries(obj)) {
+            result[key] = processParams(value);
+          }
+          return result;
+        }
+        if (typeof obj === 'string') {
+          if (obj.startsWith('0x')) {
+            return truncateHex(obj);
+          }
+        }
+        return obj;
+      };
+
+      // Process the entire request body
+      const processed = processParams(parsed);
+      return JSON.stringify(processed, null, 2);
+    } catch {
+      return body;
+    }
+  };
+
+  const truncateHexStringsInText = (text: string): string => {
+    const hexRegex = /0x[a-fA-F0-9]{10,}/g;
```

Comment:
```suggestion
    const hexRegex = /\b0x[a-fA-F0-9]{10,}/g;
```

```
> /0x[a-fA-F0-9]/.test(/640x480/)
true
> /\b0x[a-fA-F0-9]/.test(/640x480/)
false
```

Just in case!